![Build Badge](https://github.com/aakashsur/docker-3d-dna/actions/workflows/continuous-integration.yml/badge.svg)

# 3d-dna in Docker

  [3d-dna](https://github.com/aidenlab/3d-dna) is a popular Hi-C scaffolding method. The container is available via Docker Hub.

## Usage

```
docker pull \
  aakashsur/3d-dna

docker run \
  --rm \
  --volume $PWD:/root/results \
  --workdir /root/results \
  aakashsur/3d-dna \
  3d-dna \
  assembly.fasta \
  merged_nodups.txt 
```

# Details

The `3d-dna` command has been placed in $PATH. Easiest way to run is to have everything in $PWD, i.e. assembly .fasta file and the merged_nodups.txt file generated by [Juicer](https://github.com/aidenlab/juicer). Juicer is also available as a [Docker container](https://github.com/theaidenlab/Juicer-Docker) released by the Aiden Lab, but at the time of writing this, I would only recommend using it to generate the merged_nodups.txt file as the container breaks going further. Here's my workflow in singularity. First generate the chromosomes sizes file they want, then make the fragments file, place your fastq files appropriately, and then - 

```
singularity pull \
  docker://aidenlab/juicer
  
singularity run \
    --bind $PWD:/data,$PWD:/juicedir \
    juicer_latest.sif \
    -g assembly_name \
    -d /data \
    -z /juicedir/assembly.fasta \
    -y /juicedir/fragments.txt \
    -p sizes.txt \
    -s MboI \
    -S early \
    -t 40 
```

Get the merged_nodups.txt file, and then - 

```
singularity run \
  --bind $PWD:/root/results \
  --workdir /root/results \
  aakashsur/3d-dna \
  3d-dna \
  assembly.fasta \
  merged_nodups.txt 
```
